src/wifi.rs

```diff
@@ -352,7 +353,13 @@ impl EspWifi {
 
         esp!(unsafe { esp_wifi_set_config(wifi_interface_t_WIFI_IF_STA, &mut wifi_config) })?;
 
-        self.set_client_ip_conf(&conf.ip_conf)?;
+        let hostname = if let Some(hn) = &conf.hostname {
```

Comment:
Thank you for your contribution and for fixing the formatting error.

If you don't mind I would request a few extra changes. See below for why and how.

1) The notion of "hostname" is not really a Wifi STA mode specific thing. It is a DHCP-client specific thing, which - by the way - should work just fine for the Eth driver as well. The `wifi::ClientConfiguration` WiFi struct is a network L2 thing (where you've put the hostname), while the `ipv4::ClientConfiguration` is a network L3 thing (IP) where the hostname actually belongs.

2) Moreover, since the `hostname` is something related to DHCP *only*, I suggest you model it as follows (this is in `embedded-svc/ipv4.rs`:
```
#[derive(Clone, Debug, PartialEq)]
#[cfg_attr(feature = "use_serde", derive(Serialize, Deserialize))]
pub struct DHCPClientSettings {
    pub hostname: Option<String>,
}

impl Default for DHCPClientSettings {
    fn default() -> Self {
        Self {
            hostname: None,
        }
    }
}

#[derive(Clone, Debug, PartialEq)]
#[cfg_attr(feature = "use_serde", derive(Serialize, Deserialize))]
pub enum ClientConfiguration {
    DHCP(DHCPClientSettings), // This is new, just like the DHCPClientSettings struct. It used to be just "DHCP,"
    Fixed(ClientSettings),
}
```

Reply:
Can't it be set for `Fixed` as well?